File: Self_Solving_Algorithm/Cube_Algorithms/Common_Moves/Sledge_Hammer.py
from ...Cube_Turning.Cube_Turning_CW import CubeArrayRCW, CubeArrayUCW
from ...Cube_Turning.Cube_Turning_CCW import CubeArrayFCCW, CubeArrayRCCW
import copy

SledgeHammerNotation = [
    CubeArrayRCW,
    CubeArrayUCW,
    CubeArrayRCCW,
    CubeArrayFCCW
]

def SledgeHammerAlgorithm(cube):
    CurrentState = copy.deepcopy(cube)
    for move in SledgeHammerNotation:
        CurrentState = move(CurrentState)
    return CurrentState

File: Algorithm/cubeAlgorithms/commonMoves/sledgeHammer.js
// Sledge hammer is the commonly seen sequence (R U R' F')

import { CubeArrayFCCW, CubeArrayRCCW } from "../../cubeTurningCCW";
import { CubeArrayRCW, CubeArrayUCW } from "../../cubeTurningCW";

var SledgeHammerNotation = [
  CubeArrayRCW,
  CubeArrayUCW,
  CubeArrayRCCW,
  CubeArrayFCCW,
];

export function SledgeHammerAlgorithm(cube) {
  let CurrentState = JSON.parse(JSON.stringify(cube));
  for (let i = 0; i < SledgeHammerNotation.length; i++) {
    CurrentState = SledgeHammerNotation[i](CurrentState);
  }

  return CurrentState;
}
